runtime/lib/ttmetal/command_queue.cpp

```diff
@@ -426,7 +426,26 @@ void CQExecutor::execute(
 void CQExecutor::execute(
     ::tt::target::metal::EnqueueWriteBufferCommand const *command) {
   ZoneScopedN("EnqueueWriteBufferCommand");
-  throw std::runtime_error("Unsupported EnqueueWriteBufferCommand");
+  assert(buffers.find(command->dst()->global_id()) != buffers.end() &&
+         "Buffer not allocated");
+  auto buffer = buffers[command->dst()->global_id()];
+  constexpr bool blocking = false;
+  switch (command->src_type()) { // currently only supporting ConstantBuffer32
+  case tt::target::metal::HostBuffer::ConstantBuffer32: {
+    auto constantBuffer32 = command->src_as_ConstantBuffer32();
+    assert(constantBuffer32->data() != nullptr &&
+           "Only constant data supported");
+    assert(command->dst()->size() ==
+               buffers[command->dst()->global_id()]->size() &&
+           "Shape mismatch");
+    std::vector<uint32_t> vec(command->dst()->size(),
```

Comment:
command->dst()->size() doesn't seem right here, I think that's in bytes, but we want flattened shape (to number of elements) conversion (ie. 32x32 = 1024, but size is 4096 here).

Reply:
I think size and dims correlate with one an other by a factor of 4. Do we want to write to memory a vector of size 1024 or 4096 (based on the example)?